core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -227,8 +226,12 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
       std::string msg = "Kokkos::RangePolicy bounds error: The lower bound (" +
                         std::to_string(m_begin) +
                         ") is greater than the upper bound (" +
-                        std::to_string(m_end) + ").";
+                        std::to_string(m_end) + ").\n";
+#if !defined(KOKKOS_ENABLE_DEPRECATED_CODE_4)
       Kokkos::abort(msg.c_str());
+#elif defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS)
+      Kokkos::Impl::log_warning(msg);
+#endif
```

Comment:
For `MDRangePolicy`, there weren't any preconditions for the bounds at all prior to the merged PR, but for `RangePolicy`, its original behavior was to silently fallback to an empty range.
If to follow through a proper deprecation cycle, should enabling `KOKKOS_ENABLE_DEPRECATED_CODE` preserve its original behavior?

Reply:
Yes it should preserve the original behavior.  I see your point.  Will fix.